### DISCUSSION TITLE
improves the performance of the ODI DFP-34X-2C2

### DISCUSSION THREAD

**Discussion by andersonviudes** [Anime4000/RTL960x#156] [General]
hey guys,
Does anyone know how to improve the performance of the ODI DFP-34X-2C2, regarding ping, packets...
I don't know if anyone has researched or studied some commands.

I tried to look for something about it, but I didn't find it, I opened this topic to deal with performance

**Comment by npdgm** [comment]
Hi,
It's using an hardware based engine for VLANs so packet processing should happen at wire speed, meaning as fast as the SFP interface could go.
At best you could get the interface to 2.5Gbps if you're still using gigabit. That won't do anything for performance unless your ISP does allow throughput above 1Gbps.

Also it may be beneficial to turn on "flow control" (tx, or both rx and tx) on the router or switch interface where the module is plugged in. Or if you have flow control globally enabled and can't control it per port, try disabling it and see how it goes. But ideally it should be active on the GPON interface (and in my opinion nowhere else for a small network).

You should tell more about those problems you have...

In my experience with one DFP-34X-2C2 unit, this product doesn't have the best optical Rx sensitivity. Look for the Rx level on the ODI web UI. If it's bellow about -30 dBm then you're in a gray zone. That's not a good level to receive but it happens with crappy PON trees.

Lastly, and you may know that already, make sure you're using the right fiber cable. DFP-34X-2C2 has an SC/PC connector, so it must receive a **blue** SC plug. If the fiber plug is **green** on the module side then it's not the right cable, green is SC/APC. Plugging an APC on a PC may damage both, and result in very poor performance.

**Comment by Gruni22** [comment]
Hi,
I know this discussion is quite old. I have a related question.
Is it somehow possible to measure the real transfer speed from the ODI Stick to a client connected to the router/switch - like iperf.
I want to verify if the SFP Link Speed is really established via HISGMII (LAN SDS MODE 4).

Thanks in advance.

**Reply by Anime4000** [reply]
you can't have iperf in the PON Stick because SoC is just RISC MIPS running at 400MHz, it will limited by CPU.
if you really need to test, get a OLT

**Reply by Gruni22** [reply]
That makes sense...
Nevertheless it might be very complicated.
Is there a way to show the current connection/sync speed on ODI stick?

**Reply by Anime4000** [reply]
custom firmware can show:
![image](https://github.com/Anime4000/RTL960x/assets/1908715/76523fa1-505c-409d-8834-d84187a6306f)

if use telnet, type:
```
# cat /proc/lan_sds/lan_sds_cfg
lan_sds_mode = 4(HiSGMII PHY)
```

**Comment by kelt84** [comment]
Hello everyone, tell me, maybe someone knows how to solve this issue, I bought a DFP-34X-2C2 and reflashed it with all available firmware, the situation is as follows, authentication with the provider is carried out by the serial number of the device, when entering the serial number and then saving, it asks to enter the PLOAM password, but since authentication is carried out by the serial number, the PLOAM field must be left empty, how can this be done?
